src/simulator_core/entities/assets/demand_cluster.py

```diff
@@ -29,50 +54,159 @@ def __init__(
         :param str asset_id: The unique identifier of the asset.
         :param PandapipesNet pandapipe_net: Pandapipes network object to register asset to.
         """
-        super().__init__(asset_name, asset_id, panda_pipes_net)
+        super().__init__(asset_name=asset_name, asset_id=asset_id, pandapipe_net=pandapipe_net)
+
+        self.setpoints = dict()
+        self.parameters = dict()
+        self.states = dict()
+
+        self.internal_diameter = DEFAULT_DIAMETER
+        self.temperature_supply = DEFAULT_TEMPERATURE
+        self.temperature_return = DEFAULT_TEMPERATURE - DEFAULT_TEMPERATURE_DIFFERENCE
+        self.temperature_return_target = self.temperature_return
+        self.pressure_input = DEFAULT_PRESSURE
+        self.thermal_power_allocation = 0
+        self.mass_flowrate = 0
+
+        # Objects of the asset
+        self._initialized = False
+        self._intermediate_junction: Optional[Junction] = None
+        self._flow_control = None
+        self._heat_exchanger = None
+        self._simulated = False
+        self._from_junction = None
+        self._to_junction = None
+        # Output list
+        self.output: list = []
 
-    def add_physical_data(self, data: Dict[str, float]) -> None:
-        """Method to add physical data to the asset.
+    def create(self) -> None:
+        """Create a representation of the asset in pandapipes.
 
-        :param dict data:dictionary containing the data to be added the asset. The key is the name
-                        of the property, the value of the dict is the value of the property.
-        :return:
-        """
-        pass
+        The DemandCluster asset contains multiple pandapipes components.
 
-    def set_setpoints(self, setpoints: Dict, **kwargs) -> None:
+        The component model contains the following components:
+        - A flow control valve to set the mass flow rate of the system.
+        - A heat exchanger to set the power consumption of the demand
+        - An intermediate junction to link both components.
+        :param pandapipesNet pandapipes_net: pandapipes network object
+        """
+        if not self._initialized:
+            # Create intermediate junction
+            self._intermediate_junction = Junction(
+                pandapipes_net=self.pandapipes_net,
+                pn_bar=self.pressure_input,
+                tfluid_k=self.temperature_supply,
+                height_m=DEFAULT_NODE_HEIGHT,
+                name=f"intermediate_junction_{self.name}",
+            )
+            # Create the control valve
+            self._flow_control = ControlValve(
+                pandapipes_net=self.pandapipes_net,
+                controlled_mdot_kg_per_s=self.mass_flowrate,
+                diameter_m=self.internal_diameter,
+                control_active=True,
+                in_service=True,
+                name=f"flow_control_{self.name}",
+            )
+            self._flow_control.from_junction = self._from_junction
```

Comment:
Change to self.from_juction form AssetAbstract

Reply:
done